Add servicenow-salesforce and shopify-salesforce samples. (#39)
ballerina-guides/integration-samples: shopify-customer-to-salesforce-customer/Readme.md

# Shopify new customer to Salesforce customer

This sample listens for notifications from Shopify on customer updates and creates or updates the corresponding customer in Salesforce.

## Use case

> It is important to synchronize customer and sales data among ecommerce platforms and Salesforce. In Shopify, a popular ecommerce platform, customers can register by providing their email addresses. Then customers can update their details later at anytime. This sample listens for customer data updates in Shopify and create or update corresponding customer details in Salesforce.

Reviewer: ```suggestion
It is important to synchronize customer and sales data among e-commerce platforms and Salesforce. In Shopify, a popular e-commerce platform, customers can register by providing their email addresses. Then, customers can update their details later at any time. This sample listens for customer data updates in Shopify and creates or updates corresponding customer details in Salesforce.
```

    *   Name - Text(80)
    *   Email - Email type
    *   Address - Long Text Area(500)

### Setting up the Shopify account
1. Create a new Shopify partner account from `https://www.shopify.com/partners`
2. Create a Shopify development store (`https://help.shopify.com/en/partners/dashboard/managing-stores/development-stores`)
3. In the development store, navigate to `Settings` -> `Notifications` -> `Webhooks` and create webhooks for `Customer creation` and `Customer update` events. Provide the URL `http://<host>:<port>/salesforce-bridge/customers` for both webhooks.

### Config.toml
```
# ==========================
# Salesforce configuration
# ==========================

[salesforceConfig]
baseUrl = "<salesforce-base-url>"

[salesforceConfig.auth]
clientId = "<salesforce-client-id>"
clientSecret = "<salesforce-client-secret>"
refreshToken = "<salesforce-refresh-token>"
refreshUrl = "<salesforce-refresh-url>"
```

## Testing

1. Navigate to the online store view of the development store and register a new customer by providing an email address. Validate the confirmation email sent to the provided email address.

> 2. Login to Salesforce and check the Customers records. New customer record with the given email will be created.

Reviewer: ```suggestion
2. Login to Salesforce and check the Customer records. A new customer record with the given email will be created.
```

> 3. In the Shopify store, change the customer's first name and last name. Customer name of the corresponding customer will be updated in Salesforce.

Reviewer: ```suggestion
3. In the Shopify store, change the customer's first name and last name. The customer name of the corresponding customer will be updated in Salesforce.
```
